Permanent/ViewModels/PublicProfilePageViewModel.swift

```diff
@@ -87,7 +87,7 @@ class PublicProfilePageViewModel: ViewModelInterface {
         }
     }
     
-    func createProfileItem(fieldname: FieldNameUI, itemValue: String, archiveId: Int?, profileItemId: Int?) -> ProfileItemVOData? {
+    func createProfileItem(itemType: ProfileItemType, itemValue: String, archiveId: Int?, profileItemId: Int?) -> ProfileItemVOData? {
```

Comment:
We should be able to send a ProfileItemModel subclass instance and get rid of the whole switching logic in this function.